great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -121,18 +120,27 @@ def _query_for_year_and_month(
     return {"year": year, "month": month}
 
 
+SQL_DATETIME_COUNTER = 0
+
+
 def _get_sql_datetime_range(
-    conn: sqlalchemy.engine.base.Connection, table_name: str, col_name: str
+    conn: sqlalchemy.engine.base.Connection, data_from: str, col_name: str
 ) -> DatetimeRange:
-    q = f"select min({col_name}), max({col_name}) from {table_name}"
+    global SQL_DATETIME_COUNTER
+    from_name = f"_gx_sql_datetime_range_data_{SQL_DATETIME_COUNTER}"
+    SQL_DATETIME_COUNTER += 1
+    q = f"select min({col_name}), max({col_name}) from ({data_from}) as {from_name}"
```

Comment:
@billdirks Should we not be building a formal `SQLAlchemy` query object for the given DBMS backend, and then utilizing something like the following:

```
            query: str = str(
                selectable.compile(
                    dialect=execution_engine.dialect,
                    compile_kwargs={"literal_binds": True},
                )
            )
```

in order to obtain the string-valued query version for printing?  Thanks.

Reply:
Can you provide a code snippet and/or point me to docs about how to create a formal `SQLALchemy` object with a subselect in the from clause?

Reply:
@billdirks Examples that come to mind are:
* In `great_expectations/expectations/metrics/map_metric_provider.py`: `_sqlalchemy_map_condition_unexpected_count_value()` and `_sqlalchemy_map_condition_unexpected_count_aggregate_fn()` (among others).
* In `great_expectations/expectations/metrics/column_aggregate_metrics/column_quantile_values.py`, methods called by `_sqlalchemy()`.

Happy to discuss.  Thank you!

Reply:
Thank you! This is done.